presto-main/src/main/java/com/facebook/presto/SystemSessionProperties.java

```diff
@@ -462,6 +469,11 @@ public static boolean isOptimizeHashGenerationEnabled(Session session)
 
     public static JoinDistributionType getJoinDistributionType(Session session)
     {
+        // distributed_join takes precedence until we remove it
+        if (!session.getSystemProperty(DISTRIBUTED_JOIN, Boolean.class)) {
```

Comment:
This is not really backwards compatible approach: someone may set `distributed_join=false` but they still won't get BROADCAST join they were asking for.

Reply:
I don't follow.  if they set distributed_Join=false it will always broadcast.  That's what this code is saying.

Reply:
forget my example, here is a new one:

Consider the system's global is to broadcast by default and someone sets `distributed_join=true` in their session because they want to have partitioned join. They won't have it, because when `distributed_join` is true, it's not obeyed, a next setting is consulted instead.

IMO it would be much easier to understand how this works, if
- `distributed_join` was initially unset
- once `distributed_join` set, it's definitive

Reply:
but they would get distributed join because that's the default for join-distribution-type.  So they only wouldn't have it if join-distribution-type was also explicitly set.  i.e. someone fixes the main config to set join-distribution-type = BROADCAST and they try to override it with distributied_join=true;.

I agree that removing backwards compatibility for session properties all of a sudden is worse than removing it for config properties, since the config properties are probably set by one person one time and so it's simpler to fix. So maybe it's worth it to maintain 100% backwards compatibility for session properties.  But it's a tradeoff.

Reply:
Which is to say I now very much agree with you and I'll get rid of the features-config level toggle :).  It will simplify the transition process to join_distribution_type if there's no weird interaction with distributed_join.